src/components/ui/coaching-session.tsx

```diff
@@ -22,29 +34,58 @@ const CoachingSession: React.FC<CoachingSessionProps> = ({
   const { setCurrentCoachingSessionId } = useCoachingSessionStateStore(
     (state) => state
   );
+  const [updateDialogOpen, setUpdateDialogOpen] = useState(false);
 
   return (
-    <Card>
-      <CardHeader className="p-4">
-        <div className="flex flex-col sm:flex-row sm:items-center justify-between gap-2">
-          <div className="space-y-1">
-            <OverarchingGoal coachingSessionId={coachingSession.id} />
-            <div className="text-sm text-muted-foreground">
-              {format(new Date(coachingSession.date), "MMMM d, yyyy h:mm a")}
+    <>
+      <Card>
+        <CardHeader className="p-4">
+          <div className="flex flex-col sm:flex-row sm:items-center justify-between gap-2">
+            <div className="space-y-1">
+              <OverarchingGoal coachingSessionId={coachingSession.id} />
+              <div className="text-sm text-muted-foreground">
+                {format(new Date(coachingSession.date), "MMMM d, yyyy h:mm a")}
+              </div>
             </div>
+            <DropdownMenu>
+              <DropdownMenuTrigger asChild>
+                <Button variant="ghost" size="icon">
+                  <MoreHorizontal className="h-4 w-4" />
+                </Button>
+              </DropdownMenuTrigger>
+              <DropdownMenuContent align="end">
+                <DropdownMenuItem asChild>
+                  <Link
+                    href={`/coaching-sessions/${coachingSession.id}`}
+                    onClick={() =>
+                      setCurrentCoachingSessionId(coachingSession.id)
+                    }
+                  >
+                    Join Session
+                  </Link>
+                </DropdownMenuItem>
+                <DropdownMenuItem onClick={() => setUpdateDialogOpen(true)}>
+                  Update Session
+                </DropdownMenuItem>
+                <DropdownMenuItem className="text-destructive">
+                  Delete Session
+                </DropdownMenuItem>
+              </DropdownMenuContent>
+            </DropdownMenu>
           </div>
-          <Link href={`/coaching-sessions/${coachingSession.id}`} passHref>
-            <Button
-              size="sm"
-              className="w-full sm:w-auto mt-2 sm:mt-0"
-              onClick={() => setCurrentCoachingSessionId(coachingSession.id)}
-            >
-              Join Session
-            </Button>
-          </Link>
-        </div>
-      </CardHeader>
-    </Card>
+        </CardHeader>
+      </Card>
+      <CoachingSessionDialog
+        open={updateDialogOpen}
+        onOpenChange={setUpdateDialogOpen}
+        onCoachingSessionUpdated={() => {
+          // Refresh the list of coaching sessions
+          window.location.reload();
```

Comment:
Is there a hook for the coaching sessions that we can call `refresh()` on instead of this? I may be confused on what this is for otherwise.